src/Features/CSharp/Portable/UseLocalFunction/CSharpUseLocalFunctionCodeFixProvider.cs

```diff
@@ -159,60 +172,80 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
-            var type = parameter?.Type.GenerateTypeSyntax() ?? s_objectType;
-
-            var parameterSyntax = SyntaxFactory.Parameter(lambdaExpression.Parameter.Identifier).WithType(type);
-            var param = delegateInvokeMethod.Parameters[0];
-            if (param.HasExplicitDefaultValue)
+            return invocation.ReplaceNodes(invocation.ArgumentList.Arguments, (argumentNode, _) =>
             {
-                parameterSyntax = parameterSyntax.WithDefault(GetDefaultValue(param));
-            }
+                if (argumentNode.NameColon == null || argumentNode.NameColon.IsMissing)
+                {
+                    return argumentNode;
+                }
+
+                var parameterIndex = TryDetermineParameterIndex(argumentNode.NameColon, method);
+                if (parameterIndex == -1)
+                {
+                    return argumentNode;
+                }
 
-            return SyntaxFactory.ParameterList(
-                SyntaxFactory.SeparatedList<ParameterSyntax>().Add(parameterSyntax));
+                var newParameter = newParameterList.Parameters.ElementAtOrDefault(parameterIndex);
+                if (newParameter == null || newParameter.Identifier.IsMissing)
+                {
+                    return argumentNode;
+                }
+
+                return argumentNode.WithNameColon(argumentNode.NameColon.WithName(SyntaxFactory.IdentifierName(newParameter.Identifier)));
+            });
         }
 
-        private ParameterListSyntax GenerateParenthesizedLambdaParameterList(
-            SemanticModel semanticModel, ParenthesizedLambdaExpressionSyntax lambdaExpression, IMethodSymbol delegateInvokeMethod, CancellationToken cancellationToken)
+        private static int TryDetermineParameterIndex(NameColonSyntax argumentNameColon, IMethodSymbol method)
         {
-            int i = 0;
-            return lambdaExpression.ParameterList.ReplaceNodes(
-                lambdaExpression.ParameterList.Parameters,
-                (parameterNode, _) =>
-                {
-                    if (parameterNode.Type != null)
-                    {
-                        return parameterNode;
-                    }
-
-                    var parameter = semanticModel.GetDeclaredSymbol(parameterNode, cancellationToken);
-                    parameterNode = parameterNode.WithType(parameter?.Type.GenerateTypeSyntax() ?? s_objectType);
-                    var param = delegateInvokeMethod.Parameters[i++];
-                    if (param.HasExplicitDefaultValue)
-                    {
-                        parameterNode = parameterNode.WithDefault(GetDefaultValue(param));
-                    }
-
-                    return parameterNode;
-                });
+            var name = argumentNameColon.Name.Identifier.ValueText;
+            return method.Parameters.IndexOf(p => p.Name == name);
```

Comment:
This logic is basically copied from [ArgumentSyntaxExtensions.DetermineParameter](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp.Workspaces/Extensions/ArgumentSyntaxExtensions.cs,46). I'm not too happy about having to reimplement the logic for C# argument resolution here, but I suppose it's fine. `DetermineParameter` also takes care of positional arguments which I don't need but more importantly it requires the semantic model, I can't just give it a symbol for it to use.

If I used it, then I'd have to either recalculate the semantic model after every change, or bring alone old nodes along with the new ones etc etc. This seemed like the simplest solution after all. Argument resolution isn't likely to change anyway.